R/transformCounts.R

```diff
@@ -307,6 +346,120 @@ setMethod("transformAssay", signature = c(x = "SummarizedExperiment"),
     return(mat)
 }
 
+####################################.calc_css###################################
+# This function applies cumulative sum scaling (CSS) to the abundance table.
+.calc_css <- function(mat, css_percentile = NULL, ...) {
+    # Check for the presence of the scaling factor percentile and scaling 
+    # constant in additional args
+    if (is.null(css_percentile)) {
+      css_percentile <- ifelse("css_percentile" %in% names(list(...)), 
+                               list(...)$css_percentile, 0.5)
+    }
+    scaling_constant <- ifelse("scaling_constant" %in% names(list(...)), 
+                               list(...)$scaling_constant, 1000)
+    
+    # Call .calc_scaling_factors method to calculate scaling factors
+    scaling_factors <- .calc_scaling_factors(mat, css_percentile)
+    
+    # Normalize the count data by dividing by the scaling factor
+    normalized_data <- sweep(mat, 2, scaling_factors / scaling_constant, "/")
+    return(normalized_data)
+}
+
+####################################.calc_css_fast##############################
+# This function applies cumulative sum scaling (CSS) to the abundance table 
+# using the fast variant method.
+.calc_css_fast <- function(mat, ...) {
+    # Check for the presence of the rel additional arg and set default if not present
+    rel <- ifelse("rel" %in% names(list(...)), list(...)$rel, 0.1)
+    
+    # Calculate the percentile using .calc_css_percentile
+    css_percentile <- .calc_css_percentile(mat, rel, ...)
+    
+    # Call .calc_css with the calculated percentile
+    normalized_data <- .calc_css(mat, css_percentile, ...)
+    return(normalized_data)
+}
+
+#################################.calc_css_percentile######################
+# Calculates the cumulative sum scaling (css) scaling percentiles from the given 
+# data
+#'  @importFrom matrixStats rowMedians
+.calc_css_percentile <- function(mat, rel, ...) {
+    # Sort the non-zero values in each column of the matrix in decreasing order
+    smat <- lapply(1:ncol(mat), function(i) {
+      sort(mat[which(mat[, i] > 0), i], decreasing = TRUE)
+    })
+    
+    # Find the maximum length of the sorted columns
+    leng <- max(sapply(smat, length))
+    
+    # Check if any column has only one or zero features and stop if true
+    if (any(sapply(smat, length) == 1)) stop("Warning: sample with one or zero features")
```

Comment:
stop and warning calls should have `call. = FALSE`. Also check that this line does not exceed 80 characters